From 7311aa260f3aeb857e42081a127924ed1d5b61af Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Fri, 7 Aug 2026 20:07:25 +0200
Subject: [PATCH 8/9] Revert "winex11: Listen for ConfigureNotify events on the
 virtual desktop."

Fixes a regression with QT menus turning invisible after interacting
with a child window inside the virtual desktop. Annoyingly enough,
there's still a chance of that occuring, but it should be a lot less
frequent now.

Source: https://gitlab.winehq.org/wine/wine/-/commit/06b1a49f
Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/winex11.drv/desktop.c | 2 +-
 dlls/winex11.drv/window.c  | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/dlls/winex11.drv/desktop.c b/dlls/winex11.drv/desktop.c
index 703ab98..7437208 100644
--- a/dlls/winex11.drv/desktop.c
+++ b/dlls/winex11.drv/desktop.c
@@ -74,7 +74,7 @@ BOOL X11DRV_CreateDesktop( const WCHAR *name, UINT width, UINT height )
     /* Create window */
     win_attr.event_mask = ExposureMask | KeyPressMask | KeyReleaseMask | EnterWindowMask |
                           PointerMotionMask | ButtonPressMask | ButtonReleaseMask | FocusChangeMask |
-                          StructureNotifyMask | PropertyChangeMask;
+                          PropertyChangeMask;
     win_attr.cursor = XCreateFontCursor( display, XC_top_left_arrow );
 
     if (default_visual.visual != DefaultVisual( display, DefaultScreen(display) ))
diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index cbc6f0c..bfece4c 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -1474,6 +1474,9 @@ static void window_set_config( struct x11drv_win_data *data, RECT rect, BOOL abo
     TRACE( "window %p/%lx, requesting config %s mask %#x above %u, serial %lu\n", data->hwnd, data->whole_window,
            wine_dbgstr_rect(new_rect), mask, above, data->configure_serial );
     XReconfigureWMWindow( data->display, data->whole_window, data->vis.screen, mask, &changes );
+
+    /* don't expect a ConfigureNotify while window is unmapped */
+    if (data->pending_state.wm_state == WithdrawnState) data->configure_serial = 0;
 }
 
 /***********************************************************************
@@ -1751,7 +1754,6 @@ static UINT window_update_client_config( struct x11drv_win_data *data )
     UINT flags;
 
     if (!data->managed) return 0; /* unmanaged windows are managed by the Win32 side */
-    if (is_virtual_desktop()) return 0; /* ignore window manager config changes in virtual desktop mode */
     if (data->desired_state.wm_state != NormalState) return 0; /* ignore config changes on invisible/minimized windows */
 
     if (data->wm_state_serial) return 0; /* another WM_STATE update is pending, wait for it to complete */
-- 
2.55.0

